### ISSUE TITLE
BERT perf regression with Transformers 5.x: TTNN SDPA verifier rejection and fallback to manual attention

### ISSUE THREAD

**Issue by ssaliceTT** [tenstorrent/tt-xla#3942]
### Describe the bug

When running BERT with the transformers 5.x uplift (`4.57.1` → `5.2.0`), throughput drops from ~65 SPS to ~27 SPS. The root cause is a shape mismatch in the SDPA fusing pass in tt-mlir.

**What changed in transformers 5.x:**
In 4.57.1, `_prepare_4d_attention_mask` helpers built the attention mask using `torch.where`, producing a fully materialized `[B, 1, S, S]` tensor (lowered to a `WhereOp` in the IR). In 5.x, the new `sdpa_mask()` function in `masking_utils.py` instead builds the mask by broadcasting index tensors via `_non_vmap_expansion_sdpa`, producing a `[B, 1, 1, S]` result, then calling `.expand(B, -1, S, S)` ([masking_utils.py:509](https://github.com/huggingface/transformers/blob/main/src/transformers/masking_utils.py#L509)) to get the logical `[B, 1, S, S]` shape. Because `.expand()` is a zero-stride broadcast, PyTorch lowers it to a `RepeatOp` in the IR rather than a `WhereOp`.

**Why the fusing pass breaks:**
`SDPAFusingPattern::prepareMask` unconditionally strips `RepeatOp`s. Thus, the seq-len dim `RepeatOp` introduced by `.expand()` also gets stripped, leaving a `[B, 1, 1, S]` tensor instead of `[B, 1, S, S]`. The TTNN SDPA op verifier then rejects it:

```
loc("dot.4870"): error: 'ttnn.scaled_dot_product_attention' op Attention mask at dim 2 must match query sequence length
```

This causes silent fallback to decomposed manual f32 attention, explaining the ~2.4× perf regression.

**Perf numbers and nightly runs for reference:**

| Scenario | SPS | Nightly Runs |
|---|---|---|
| Before uplift (transformers 4.57.1) | ~32.99 | [Nightly #496](https://github.com/tenstorrent/tt-xla/actions/runs/23222601518/job/67526606911#step:15:94) |
| After uplift (transformers 5.2.0) | ~27.15 | [Nightly #497](https://github.com/tenstorrent/tt-xla/actions/runs/23273415455/job/67717732800#step:15:526)
| After revert to main | ~65.20 | [Nightly #500](https://github.com/tenstorrent/tt-xla/actions/runs/23367591831/job/67994213663#step:15:94)

**Possible fixes (both verified locally):**
1. **tt-mlir fix**: In `SDPAFusingPattern::prepareMask`, guard `RepeatOp` stripping on the seq-len dim. Only strip if `dims[kSeqLenDim] == 1` (i.e. it's a head-dim repeat, not a seq-len repeat). This mirrors the existing guard in `analyzeQ` for `RepeatInterleaveOp`.
2. **Front-end workaround**: Wrap `BertModel` to pre-materialize the mask via `torch.where` before calling `forward`, bypassing `sdpa_mask()` entirely and matching the 4.57.1 construction pattern. Check this branch for reference [ssalice/bert-regression-fix](https://github.com/tenstorrent/tt-forge-models/compare/main...ssalice/bert-regression-fix).

Likely better fixes and I'm bit out of my depth with the tt-mlir one, but thought I'd add them anyways.

Related tt-mlir PR that addressed the correctness side of this (but not the perf regression): https://github.com/tenstorrent/tt-mlir/pull/7020

### Steps to reproduce the issue

Note: Either work off of uplift branch `ssalice/transformers-5.2-reuplift` or if reuplift is complete work off of main.

1. Run BERT perf benchmark on n150:
```bash
pytest -svv tests/benchmark/test_encoders.py::test_bert
```
There should be a blurb that pops up like so:
```====================================================================
| pytorch_BERT_emrecan/bert-base-turkish-cased-mean-nli-stsb-tr_nlp_embed_gen_huggingface Benchmark Results:|
--------------------------------------------------------------------
| Model: pytorch_BERT_emrecan/bert-base-turkish-cased-mean-nli-stsb-tr_nlp_embed_gen_huggingface
| Model type: Encoder, Text Embedding
| Dataset name: Benchmark Sentences
| Date: 19-03-2026
| Machine name: eb3f6f8161d0
| Total execution time: 9.425938165
| Total samples: 256
| Sample per second: 27.15910029524366
| Evaluation score: 0.9968794584274292
| Batch size: 8
| Data format: bfloat16
| Input sequence length: 384
====================================================================
PASSED
```
2. Compare samples per second against nightly runs
